src/Agent.Listener/DistributedTask.Pipelines/Yaml/PipelineParser.cs

```diff
@@ -53,52 +72,57 @@ public Process Load(String defaultRoot, String path, IDictionary<String, Object>
             ResolveTemplates(process, defaultRoot: processFile.Directory, cancellationToken: cancellationToken, fileCount: ref fileCount);
 
             // Create implied levels for the process.
-            if (process.Jobs != null)
-            {
-                var newPhase = new Phase { Jobs = process.Jobs, Name = process.Name };
-                process.Phases = new List<IPhase>();
-                process.Phases.Add(newPhase);
-                process.Jobs = null;
-            }
-            else if (process.Steps != null)
+            if (process.Steps != null)
             {
-                var newJob = new Job { Steps = process.Steps, Name = process.Name, Variables = process.Variables };
-                var newPhase = new Phase { Jobs = new List<IJob>() };
-                newPhase.Jobs.Add(newJob);
+                var newPhase = new Phase { Steps = process.Steps, Name = process.Name, Variables = process.Variables };
                 process.Phases = new List<IPhase>();
                 process.Phases.Add(newPhase);
                 process.Steps = null;
             }
 
-            var knownPhaseNames = new HashSet<String>(StringComparer.OrdinalIgnoreCase);
-            var knownJobNames = new HashSet<String>(StringComparer.OrdinalIgnoreCase);
+            // Convert "checkout" steps into variables.
```

Comment:
Right now I am converting `checkout` into variables. I wanted to make `checkout` work in yaml today. In a future version, we need to change this up so that an actual checkout step is sent down in the job message.